cockatrice/src/dlg_settings.cpp

```diff
@@ -543,6 +510,91 @@ DeckEditorSettingsPage::DeckEditorSettingsPage()
     setLayout(lpMainLayout);
 }
 
+/**
+ * If reset, this method contains the default URLs we will populate
+ */
+void DeckEditorSettingsPage::resetDownloadedURLsButtonClicked()
+{
+    urlList->clear();
+
+    urlList->addItem("https://api.scryfall.com/cards/!uuid!?format=image");
+    urlList->addItem("https://api.scryfall.com/cards/multiverse/!cardid!?format=image");
+    // urlList->addItem("http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!cardid!&type=card");
+    // urlList->addItem("http://gatherer.wizards.com/Handlers/Image.ashx?name=!name!&type=card");
+
+    storeSettings();
+    QMessageBox::information(this, tr("Success"), tr("Download URLs have been reset."));
+}
+
+void DeckEditorSettingsPage::clearDownloadedPicsButtonClicked()
+{
+    QString picsPath = settingsCache->getPicsPath() + "/downloadedPics/";
+    QStringList dirs = QDir(picsPath).entryList(QDir::AllDirs | QDir::NoDotAndDotDot);
+    bool outerSuccessRemove = true;
+    for (int i = 0; i < dirs.length(); i++) {
+        QString currentPath = picsPath + dirs.at(i) + "/";
+        QStringList files = QDir(currentPath).entryList(QDir::Files);
+        bool innerSuccessRemove = true;
+        for (int j = 0; j < files.length(); j++) {
+            if (!QDir(currentPath).remove(files.at(j))) {
+                qDebug() << "Failed to remove " + currentPath.toUtf8() + files.at(j).toUtf8();
+                outerSuccessRemove = false;
+                innerSuccessRemove = false;
+            }
+        }
+
+        if (innerSuccessRemove) {
+            bool success = QDir(picsPath).rmdir(dirs.at(i));
+            if (!success) {
+                qDebug() << "Failed to delete inner directory" << picsPath;
+            }
+        }
+    }
+    if (outerSuccessRemove)
+        QMessageBox::information(this, tr("Success"), tr("Downloaded card pictures have been reset."));
+    else
+        QMessageBox::critical(this, tr("Error"), tr("One or more downloaded card pictures could not be cleared."));
+}
+
+void DeckEditorSettingsPage::actAddURL()
+{
+    bool ok;
+    QString msg = QInputDialog::getText(this, tr("Add URL"), tr("URL:"), QLineEdit::Normal, QString(), &ok);
+    if (ok) {
+        urlList->addItem(msg);
+        storeSettings();
+    }
+}
+
+void DeckEditorSettingsPage::actRemoveURL()
+{
+    if (urlList->currentItem()) {
```

Comment:
Is this right? You say if item, delete current row.

Looking at the docs, it seems like `currentItem` returns a pointer - can you do an explicit `nullptr` comparison instead of using it as bool-ish?

Reply:
So I took this from the macros code exactly as it was. But sure, I can make it `!= nullptr` if that's what you'd prefer